### PR TITLE
new changes from PLP v2

### PR DESCRIPTION
**Pull request by NadiyaS**
## Description

This PR contain the latest changes of PLP, which were released in [version 2](https://plp-widgets-ui.magento-ds.com/v2/search.js) for LS 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: dev-template.html [hunk 1]
```diff
@@ -52,10 +52,10 @@
   <script type="text/javascript">
     const root = document.querySelector('div.search-plp-root');
     const storeDetails = {
-      environmentId: '',
-      websiteCode: '',
-      storeCode: '',
-      storeViewCode: '',
+      environmentId: 'cf1d5897-72e8-471a-8c2f-2bed259de3ff',
```
Comment (benjaminkalk) [inline dev-template.html:55]: Should we putting an environment Id into public repo?

File: dev-template.html [hunk 2]
```diff
@@ -65,41 +65,28 @@
...
-        // optimizeImages: true, // optimize images with Fastly
-        // imageBaseWidth: 200,
-        // resolveCartId?: resolveCartId, // Luma specific addToCart method. Enabled with the extension
-        // refreshCart?: refreshCart, // Luma specific addToCart method. Enabled with the extension
-        // addToCart?: (sku, options, quantity)=>{} // custom add to cart callback function. Called on addToCart action
+        allowAllProducts: true,
       },
       context: {
-        customerGroup: '',
+        customerGroup: 'b6589fc6ab0dc82cf12099d1c2d40ab994e8410c',
```
Comment (benjaminkalk) [inline dev-template.html:73]: Same question regarding if we should be including these keys in public repo.

File: dev-template.html [hunk 3]
```diff
@@ -123,19 +110,19 @@
...
+      storeCode: 'main_website_store',
       storeViewId: 123456,
-      storeViewCode: '',
-      websiteName: '',
-      storeName: '',
-      storeViewName: '',
+      storeViewCode: 'default',
+      websiteName: 'magento',
+      storeName: 'magento',
+      storeViewName: 'plp widget',
```
Comment (benjaminkalk) [inline dev-template.html:125]: Same comment as above.  Should we be adding environmentIds

File: src/context/store.tsx [hunk 4]
```diff
@@ -74,10 +74,12 @@ const StoreContextProvider = ({
         userViewHistory: context?.userViewHistory ?? [],
       },
       apiUrl: environmentType?.toLowerCase() === 'testing' ? TEST_URL : API_URL,
-      apiKey:
-        environmentType?.toLowerCase() === 'testing' && !apiKey
-          ? SANDBOX_KEY
-          : apiKey,
+      apiKey: environmentType?.toLowerCase() === 'testing' ? API_KEY : apiKey,
+      //This is for the publix repo setup
```
Comment (benjaminkalk) [inline src/context/store.tsx:78]: Should we keep this comment?

**Review by benjaminkalk — CHANGES_REQUESTED** [review]
I think we should remove any environment Ids, keys, or config we don't want to expose in public repo.

**Review by benjaminkalk — APPROVED** [review]

File: src/index.tsx [hunk 5]
```diff
@@ -49,21 +52,31 @@ const LiveSearchPLP = ({ storeDetails, root }: MountSearchPlpProps) => {
...
-                <CartProvider>
-                  <App />
-                </CartProvider>
-              </ProductsContextProvider>
-            </Translation>
-          </Resize>
-        </SearchProvider>
-      </AttributeMetadataProvider>
-    </StoreContextProvider>,
+    <SentryProvider>
```
Comment (grahamcrackers) [inline src/index.tsx:55]: We should remove Sentry and probably Floodgate as well.

File: src/context/sentry.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
+/*
```
Comment (grahamcrackers) [inline src/context/sentry.tsx:1]: i'm not sure if we want to expose this publicly

File: src/components/WishlistButton/WishlistButton.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';
```
Comment (grahamcrackers) [inline src/components/WishlistButton/WishlistButton.stories.mdx:1]: you are adding a .mdx file for storybook, but removing the "mdx" option in the [storybook config](https://github.com/adobe/storefront-product-listing-page/pull/27/files#diff-54238bc08b5014b3bff3682beb27adb9a1463aa0e299333a2abc50909189d9d8R6)?

**Review by grahamcrackers — CHANGES_REQUESTED** [review]
Summary:
- Storybook isn't working. I tried to run it locally and styles were broken.
- `npm run lint` exposes 23 warnings which need to be fixed
- We shouldn't be tying our Sentry and Floodgate to an open source repo. Not that it exposes anything critical, but that it will most likely cause issues if a client is forking this repo and customizing things on their end.

**Comment by NadiyaS** [comment]
Changed will be delivered in https://github.com/adobe/storefront-product-listing-page/pull/30

**Pull request closed without merging**